routes/articles.js

```diff
@@ -9,26 +9,45 @@ router.post('/', async (req,res) => {
         title : req.body.title,
         description :  req.body.description,
         author : req.body.author,
-        tags :  req.body.tags
+        tags :  req.body.tags,
+        time : req.body.time
     })
     try {
         const newArticle = await article.save()
         //201 cause its sucess
         res.status(201).json(newArticle)
-    } catch(arr) {
-        res.status(400).json( { message : err.message })
+    } catch(err) {
+        //client's mistake
+        res.status(401).json( { message : err.message })
     }
 })
 
 
 //get all articles
 router.get('/' , async (req,res) => {
     try {
-        //todo--> add more serch and filter options
-        const articles = await Article.find()
+        //todo--> add article limits
+        let searchOptions = {}
+        if( req.query.title ) {
+            searchOptions.title = new RegExp(req.query.title, 'i')
+        }
+        if( req.query.author ) {
+            searchOptions.author = new RegExp(req.query.author, 'i')
+        }
+
+        //we can have some default values of these as well
+        if( req.query.toDate && req.query.fromDate ) {
```

Comment:
if only fromDate is given then take currentDate as toDate

Reply:
if it is possible to get the least date then we can set the ```fromDate``` to that...
I'll fix for ```toDate```

Reply:
Not need of handling empty fromDate 

A user should always pass `fromDate` , Otherwise it doesn't make any sense 

So let's not complicate things